torch/csrc/jit/passes/lower_linear_block.cpp

```diff
@@ -0,0 +1,31 @@
+#include "torch/csrc/jit/passes/lower_linear_block.h"
+
+namespace torch { namespace jit {
+
+void LowerLinearBlock(Graph& g) {
+  for(auto it = g.nodes().begin(); it != g.nodes().end(); ++it) {
+    if(it->kind() == prim::LinearBlock) {
+      // if any_defined(inputs):
+      //  outputs = <origal_computation>
+      // else:
+      //  outputs = undefineds
+      WithInsertPoint guard(*it);
+      auto cond = g.insertNode(g.create(prim::AnyDefined, it->inputs()));
+      auto if_stat = g.insertNode(g.create(prim::If,{cond->output()}, it->outputs().size()));
+      if_stat->addBlock()->cloneFrom(
+          it->blocks().at(0), [](Value* v) { return v; });
```

Comment:
Is this value map correct? You can't reuse value pointers form the old block, which will get deleted in a moment

Reply:
cloneFrom only calls the map function for values not already defined in the block being copied.

Reply:
Ok, but this doesn't convince me that this function is what we want. If you know that it should never be called, then raise an error inside.

Reply:
It can be called, though. There are values that it uses which are not defined in the block but are outputs of earlier AutogradAdd or GradOf block